src/syntax_mapping.rs

```diff
@@ -122,8 +122,23 @@ impl<'a> SyntaxMapping<'a> {
             .insert("*.hook", MappingTarget::MapTo("INI"))
             .unwrap();
 
-        if let Some(xdg_config_home) = std::env::var_os("XDG_CONFIG_HOME") {
-            let git_config_path = Path::new(&xdg_config_home).join("git");
+        // global git config files
+        // `$XDG_CONFIG_HOME/git`, or `$HOME/.config/git` if `$XDG_CONFIG_HOME` is not set or empty
+        // see https://git-scm.com/docs/git-config#FILES
+        // we cover both cases regardless of the state of `$XDG_CONFIG_HOME`
+        if let Some(xdg_config_home) =
+            std::env::var_os("XDG_CONFIG_HOME").and_then(|val| (!val.is_empty()).then(|| val))
+        {
+            insert_git_config_global(&mut mapping, &xdg_config_home);
+        }
+        if let Some(default_config_home) =
+            std::env::var_os("HOME").map(|home| Path::new(&home).join(".config"))
```

Comment:
Shouldn't you check `!val.is_empty()` also here? Maybe you can move that check into the helper function?

Reply:
I mean, if `$HOME == ""` I think we can safely call the environment broken. But eh, why not 🤷.

I kept the check outside of the helper function because it's rather inconvenient to make it accept an `OsStr`. Also I realised `Option::filter` exists and therefore is best.